core/src/components/datetime/test/disabled/datetime.spec.tsx

```diff
@@ -0,0 +1,37 @@
+import { h } from '@stencil/core';
+import { newSpecPage } from '@stencil/core/testing';
+
+import { Datetime } from '../../../datetime/datetime';
+
+describe('ion-datetime disabled', () => {
+  beforeEach(() => {
+    // IntersectionObserver isn't available in test environment
+    const mockIntersectionObserver = jest.fn();
+    mockIntersectionObserver.mockReturnValue({
+      observe: () => null,
+      unobserve: () => null,
+      disconnect: () => null,
+    });
+    global.IntersectionObserver = mockIntersectionObserver;
+  });
+
+  it('picker should be disabled in prefer wheel mode', async () => {
+    const page = await newSpecPage({
+      components: [Datetime],
+      template: () => (
+        <ion-datetime id="inline-datetime-wheel" disabled prefer-wheel value="2022-04-21T00:00:00"></ion-datetime>
+      ),
+    });
+
+    await page.waitForChanges();
+
+    const datetime = page.body.querySelector('ion-datetime')!;
+    const columns = datetime.shadowRoot!.querySelectorAll('ion-picker-column-internal');
+
+    await expect(columns.length).toEqual(4);
+
+    columns.forEach((column) => {
+      expect(column.getAttribute('disabled')).not.toBeNull();
```

Comment:
I don't understand why we use `toBeNull` here if the property type is `boolean | undefined`. If `disabled={undefined}` then this check would still pass. Wouldn't we want `toBeUndefined` instead?

Reply:
The problem is you aren't including the picker components in your test, so they aren't getting defined. If you log each `column` in the `forEach` loop you'll see that the result is an empty object (`{}`).

Your `hasAttribute` works because even if the component isn't defined, the DOM node is still there which _does_ have a disabled attribute.

`getAttribute` should never return `undefined` according to https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute which is why your `toBeUndefined` check was not working with `getAttribute`.

So you could still use `toBeNull` and be fine, but I think this is confusing because when thinking about the `disabled` property it can never be null.

------

I think it would be more clear to do the following:

1. Import and pass the picker components to the `components` array.

```diff
+ import { PickerColumnInternal } from '../../../picker-column-internal/picker-column-internal';
+ import { PickerInternal } from '../../../picker-internal/picker-internal';

...

- components: [Datetime],
+ components: [Datetime, PickerColumnInternal, PickerInternal],
```


2. Check the component property instead of the attribute:

```diff
- expect(column.hasAttribute('disabled')).toBe(true);
+ expect(column.disabled).toBe(true);
```